nixos/modules/services/web-apps/baget.nix

```diff
@@ -0,0 +1,104 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.baget;
+
+  defaultConfig = {
+    "PackageDeletionBehavior" = "Unlist";
+    "AllowPackageOverwrites" = false;
+
+    "Database" = {
+      "Type" = "Sqlite";
+      "ConnectionString" = "Data Source=baget.db";
+    };
+
+    "Storage" = {
+      "Type" = "FileSystem";
+      "Path" = "";
+    };
+
+    "Search" = {
+      "Type" = "Database";
+    };
+
+    "Mirror" = {
+      "Enabled" = false;
+      "PackageSource" = "https://api.nuget.org/v3/index.json";
+    };
+
+    "Logging" = {
+      "IncludeScopes" = false;
+      "Debug" = {
+        "LogLevel" = {
+          "Default" = "Warning";
+        };
+      };
+      "Console" = {
+        "LogLevel" = {
+          "Microsoft.Hosting.Lifetime" = "Information";
+          "Default" = "Warning";
+        };
+      };
+    };
+  };
+
+  configAttrs = recursiveUpdate defaultConfig cfg.extraConfig // { "ApiKey" = "REPLACE_WITH_API_KEY"; };
+
+  configFile = pkgs.writeText "appsettings.json" (builtins.toJSON configAttrs);
+
+in
+{
+  options.services.baget = {
+    enable = mkEnableOption "BaGet web application.";
+
+    apiKeyFile = mkOption {
+      type = types.path;
+      example = "/root/baget.key";
+      description = ''
+        Private API key for BaGet.
+      '';
+    };
+
+    extraConfig = mkOption {
+      type = types.attrs;
+      default = {};
+      example = {
+        "Database" = {
+          "Type" = "PostgreSql";
+          "ConnectionString" = "Server=/run/postgresql;Port=5432;";
+        };
+      };
+      description = ''
+        Extra configuration options for BaGet. Refer to <link xlink:href="https://loic-sharma.github.io/BaGet/configuration/"/> for details.
+      '';
+    };
+  };
+
+  # implementation
+
+  config = mkIf cfg.enable {
+
+    systemd.services.baget = {
+      description = "BaGet server";
+      wantedBy = [ "multi-user.target" ];
+      wants = [ "network-online.target" ];
+      after = [ "network.target" "network-online.target" ];
+      serviceConfig = {
+        ExecStart = "${pkgs.baget}/bin/BaGet";
+        WorkingDirectory = "/var/lib/baget";
+        DynamicUser = true;
+        PermissionsStartOnly = true;
```

Comment:
this was deprecated a while ago in favor of permission specifiers in `Exec*` lines. don't think there's a way to generate those with the systemd module though.

Reply:
Is there any other workaround? I could use a separate service unit that runs as `root` to prepare config beforehand, but it's so much more verbose.

Reply:
one of the permission specifiers does just that; prefix a line with `+` and it'll run with full root privileges. you could eg
```nix
ExecStartPre = "+${writeScript "baget-pre-start" '' ... ''}"
```
there's a couple more that do other (more restricted) things, describe in the `systemd.service` manpage under ExecStart.

Reply:
Oh, so we do have an ability to use them! Cool, will do just that.

Reply:
yup! we just don't have the ability to use them with the `preStart` attribute (or not that we know of at least) :)